src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -44,7 +44,8 @@ public User add(User user) {
     public Optional<User> findByEmail(String email) {
         try (Session session = sessionFactory.openSession()) {
             Query<User> findUserByEmail
-                    = session.createQuery("FROM User m where m.email = :email", User.class)
+                    = session.createQuery("FROM User u LEFT JOIN FETCH u.roles "
```

Comment:
why do we use LEFT JOIN here? is it possible to get user without role?

Reply:
fixed